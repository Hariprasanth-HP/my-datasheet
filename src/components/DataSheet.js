import * as React from "react";
import { render } from "react-dom";
import { ReactGrid, Column, Row, Highlight } from "@silevis/reactgrid";
import "@silevis/reactgrid/styles.css";
import { useEffect } from "react";
import { useState } from "react";
import { useCustomHook } from "./CustomHook";
const getPeople = () => [
  { name: "Thomas", surname: "Goldman" },
  { name: "Susie", surname: "Quattro" },
  { name: "", surname: "" },
];
const myData = [
  {
    name: "hari",
    order: 1,
    published: "2023-01-25T12:00:00Z",
    expiry: "2023-03-27T12:00:00Z",
  },
  {
    name: "yogesh",
    order: 2,
    published: "2023-02-25T12:00:00Z",
    expiry: "2023-04-27T12:00:00Z",
  },
  {
    name: "mani",
    order: 3,
    published: "2023-01-25T12:00:00Z",
    expiry: "2023-01-31T12:00:00Z",
  },
];

const getColumns = () => [
  { columnId: "name", width: 150 },
  { columnId: "surname", width: 150 },
];

const headerRow = {
  rowId: "header",
  cells: [
    { type: "header", text: "Name" },
    { type: "header", text: "Surname" },
  ],
};

const getRows = (people) => [
  headerRow,
  ...people.map((person, idx) => ({
    rowId: idx,
    cells: [
      { type: "text", text: person.name },
      { type: "text", text: person.surname },
    ],
  })),
];

const highlights = [
  { columnId: "name", rowId: 0, borderColor: "#00ff00" },
  { columnId: "surname", rowId: 1, borderColor: "#0000ff" },
  { columnId: "name", rowId: 2, borderColor: "#ff0000" },
];
export function DataSheet() {
  const [people] = React.useState(getPeople());
  const rows = getRows(people);
  const columns = getColumns();
  const [myfil, setMyfil] = useState([]);
  const [data] = useCustomHook("https://jsonplaceholder.typicode.com/todos/1");
  console.log("data", data);
  useEffect(() => {
    const filtered = myData.filter((d) => {
      var publishDate = d.published.slice(0, 10);
      var expiryDate = d.expiry.slice(0, 10);
      var CurrentDate = new Date();
      publishDate = new Date(publishDate);
      expiryDate = new Date(expiryDate);
      return (
        publishDate && CurrentDate > publishDate && CurrentDate < expiryDate
      );
    });
    setMyfil(filtered);
  }, []);
  console.log("myfile", myfil);
  return (
    <div>
      <ReactGrid rows={rows} columns={columns} highlights={highlights} />;
    </div>
  );
}
