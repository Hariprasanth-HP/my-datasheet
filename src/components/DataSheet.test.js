// Unit Test Code
import React from "react";
import { render, screen } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import DataSheet from "./DataSheet";
import ReactDOM from "react-dom";
describe("DataSheet Component", () => {
  // it("should render the component", () => {
  //   const wrapper = mount(<DataSheet />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("should render the tab container", () => {
  //   const wrapper = shallow(<DataSheet />);
  //   expect(wrapper.find('[data-testid="tab_container"]').length).toBe(1);
  // });

  // it("should render the select element", () => {
  //   const wrapper = shallow(<DataSheet />);
  //   expect(wrapper.find("select").length).toBe(1);
  // });

  // it("should render the table element", () => {
  //   const wrapper = shallow(<DataSheet />);
  //   expect(wrapper.find("table").length).toBe(1);
  // });

  // it("should render the ul element", () => {
  //   const wrapper = shallow(<DataSheet />);
  //   expect(wrapper.find("ul").length).toBe(1);
  // });

  // it("should render the p element with para prop", () => {
  //   const para = "This is a sample para"; // passing props to component     // mounting component with props     // finding p tag with data-testid="para"     // checking if text matches with passed props      const wrapper = mount (< DataSheet para={para}/> );      expect (wrapper.find ('[data-testid="para"]').text()).toEqual (para) ; }) ;

  //   it("should render the FormControl component ", () => {
  //     const wrapper = shallow(<DataSheet />);
  //     expect(wrapper.find("FormControl").length).toBe(1);
  //   });

  //   it("should render the InputBase component ", () => {
  //     const wrapper = shallow(<DataSheet />);
  //     expect(wrapper.find("InputBase").length).toBe(1);
  //   });
  // });
  test("shukd render", async () => {
    render(<DataSheet />);
    screen.debug();
  });
});
