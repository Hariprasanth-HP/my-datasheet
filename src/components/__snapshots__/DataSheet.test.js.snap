// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSheet Component should render the component 1`] = `
<div
  className="tab_container"
  data-testid="tab_container"
>
  <ForwardRef
    className="response_row"
    columns={
      Array [
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "name",
            "original": Object {
              "columnData": Object {
                "alignRight": false,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "name",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "ID",
            "original": Object {
              "columnData": Object {
                "alignRight": true,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "ID",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "image",
            "original": Object {
              "columnData": Object {
                "alignRight": false,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "image",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "Designation",
            "original": Object {
              "columnData": Object {
                "alignRight": false,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "Designation",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "Team",
            "original": Object {
              "columnData": Object {
                "alignRight": false,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "Team",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
        Object {
          "cellClassName": undefined,
          "columnData": Object {
            "key": "Manager",
            "original": Object {
              "columnData": Object {
                "alignRight": true,
                "continuousUpdates": true,
                "formatBlurredInput": [Function],
                "formatInputOnFocus": [Function],
                "parseUserInput": [Function],
                "placeholder": undefined,
              },
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
              "copyValue": [Function],
              "deleteValue": [Function],
              "isCellEmpty": [Function],
              "pasteValue": [Function],
            },
          },
          "component": [Function],
          "copyValue": [Function],
          "deleteValue": [Function],
          "disabled": undefined,
          "id": "Manager",
          "isCellEmpty": [Function],
          "pasteValue": [Function],
        },
      ]
    }
    onChange={[Function]}
    rowClassName={[Function]}
    value={Array []}
  />
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
  </select>
  <table
    className="grid"
  >
    <tbody>
      <tr
        className="grid-row"
      >
        <td>
          hid
        </td>
        <td>
          yes
        </td>
      </tr>
    </tbody>
  </table>
  <ul
    role="fruitss"
  >
    <li
      key="Bananas"
    >
      Bananas
    </li>
    <li
      key="Apples"
    >
      Apples
    </li>
    <li
      key="Strawberries"
    >
      Strawberries
    </li>
    <li
      key="Grapes"
    >
      Grapes
    </li>
    <li
      key="Oranges"
    >
      Oranges
    </li>
  </ul>
  <p
    data-testid="para"
  />
  <ForwardRef(FormControl)>
    <ForwardRef(InputBase)
      id="myInput"
    >
      selectyou
    </ForwardRef(InputBase)>
    <ForwardRef(Select)
      data-testid="myInputselect"
      onChange={[Function]}
      value=""
    >
      <ForwardRef(MenuItem)
        value="Bananas"
      />
    </ForwardRef(Select)>
  </ForwardRef(FormControl)>
</div>
`;
